test/unit/math/prim/fun/one_hot_vector_test.cpp

```diff
@@ -0,0 +1,38 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathFunctions, one_hot_vector) {
+  using Eigen::VectorXd;
+  using stan::math::one_hot_vector;
+
+  int K = 4;
+  VectorXd u1 = one_hot_vector(K, 1);
+  VectorXd v1(K);
+  v1 << 1, 0, 0, 0;
+
+  EXPECT_EQ(K, u1.size());
+  for (int i = 0; i < K; i++) {
+    EXPECT_EQ(u1[i], v1[i]);
+  }
+
+  int k = 4;
+  VectorXd u2 = one_hot_vector(K, k);
+  VectorXd v2(K);
+  v2 << 0, 0, 0, 1;
+
+  EXPECT_EQ(k, u2.size());
+  for (int i = 0; i < k; i++) {
+    EXPECT_EQ(u2[i], v2[i]);
+  }
+}
+
+TEST(MathFunctions, one_hot_vector_throw) {
+  using stan::math::one_hot_vector;
+  int K = 5;
```

Comment:
This needs a K = 0, k = 1 test to make sure that K = 0 is being caught as an exception.  Or the tested K needs to be put in a loop fro 0 to 5 or something.